### ISSUE TITLE
Make reaction stores reactor agnostic

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/hooks-api#114]
Strange as this might sound, it's something that has been brought up in the discussion surrounding #109 and #113. I think maybe we should give it a try and see how things turn out.

**Comment by JDGrimes** [comment]
See https://github.com/WordPoints/hooks-api/issues/109#issuecomment-194460928 especially.

## Hook Modes

The first issue that I run into is that we've put a lot of work into hook modes and making sure that the reaction stores are only made available in the correct contexts, and are automatically accessible via a generic property based on the current mode. But this is handling that takes place within the reactor, and it is based upon the reaction stores being in a children registry with the reactors as the parents and the modes as the children that each reaction store is specifically identified by. But with the stores no longer grouped per-reactor, this would have to change. To continue providing this automatic handling so that reaction stores don't have to be referenced explicitly based on hook mode, we'd have to group them in some way. I think what we could do is have the different slugs of the different hook modes as the parents, and the slugs of the different "implementations" or "use cases" or whatever we want to call them (how about "stores"?), as the children. So we'd do the same thing that we do at present, except instead of auto-filling the child slug we'd auto-fill the parent.

**Comment by JDGrimes** [comment]
Actually, I guess maybe we could do it either way, and it would work with the modes as children and the implementation slugs as the parents. But I guess it might actually be useful to have the modes as parents, as suggested above, because then we could get all stores of a particular mode, whereas getting all stores for a particular implementation isn't so useful when they would likely be mostly out of context and inaccessible. And if we want the "implementation" slug to be the store slug, it would have to be the child anyway, so I guess that settles it.

And really it shouldn't be much harder to get the all of the stores for a particular implementation, because we make it easy to get a list of all of the registered slugs in a registry.

**Comment by JDGrimes** [comment]
## `get_all_*()` methods

These reactor methods are no longer used, so I'm removing them. The `get_all_reactions()` method would still be used in the router, but we'll sort of just be moving the logic to there. If we need this sort of logic anywhere else, we'll reintroduce these methods somewhere.

**Comment by JDGrimes** [comment]
## `get_reaction_store()` method

This reactor method isn't as necessary as it once was, because the reaction stores no longer have to be constructed with a reactor object (which is the main reason that we have this method). Previously it wasn't possible to just access the reaction stores directly from the registry (or it wasn't advised). Another reason for that was that the reaction stores shouldn't really be used out of context, because this can lead to unexpected behavior. So the other important thing that method does is deny access to out-of-context reaction stores. For that reason, I think it would be good to keep this method around. For now I'll move it to the main `Hooks` class.

**Comment by JDGrimes** [comment]
This leaves us with the feeling that the hook mode API is half-baked, and that maybe hook modes should be registered, or should be more tightly coupled with the entity context API, or removed in favor of that or something. But I guess we're really done here and that would be for another ticket.

**Comment by JDGrimes** [comment]
I think this was basically completed with the above commits, the only reason it wasn't closed was that it was done in parts and we forgot to say that it was fixed when we were done (we were waiting to come back and close it I guess until the commits had actually been synced up, which took quite some time). 

Anyway, I think the only point that we weren't entirely sure about here was whether the modes should be parents or children in the registry, and we went with them being the parents and that seems to be working, so I see no need to change this. We're done here.
